### ISSUE TITLE
problem of running mimic_test.py

### ISSUE THREAD

**Issue by Kingofolk** [Holipori/EKAID#2]
Hi! Sorry for bothering you again.  I encoountered some problems when running the 'python mimic_test.py _path to checkpoint_' command. The error is as following:
![image](https://github.com/Holipori/EKAID/assets/37804732/0ec55cf4-646e-437b-b88b-558bfd87e73a)
Should I add some lines to unify their type in the following section or could you please offer me the optimal solution?
![image](https://github.com/Holipori/EKAID/assets/37804732/1e12de53-99c8-4b10-8124-3afe81cdf9ea)

Besides, I do not know whether this problem is due to the inproporiate package version. I also have problems in installing the required libs from the requirement.txt. Like this:
![image](https://github.com/Holipori/EKAID/assets/37804732/0313ee83-3815-4786-8dfa-a26a8f0301ae)
So I installed all the needed libs manually.
Looking forward to your reply. Thank you very much and wich you have a nice day.
Kingofolk

**Comment by Holipori** [comment]
Hi, sorry for the late reply. This problem is triggered because of the type miss match between the string of "2048 +4" and the integer of 2052. There should be a function using a package called "ast" in the code that should have addressed the type issue well. But it fails, most likely due to the version problem, the one I'm using is "typed-ast==1.5.1".  

Alternatively, you can fix the issue by changing “2048 + 4” in the dynamic_change_pos_mimic.yaml file to “2052” and “1024 + 2” to “1026”. This should resolve the problem.

I have also committed the fixed version to the repository.

**Comment by Kingofolk** [comment]
Thank you for your reply. When I run mimic_test.py. A new error happened.
![image](https://github.com/Holipori/EKAID/assets/37804732/afeee652-d600-4684-afe1-f91a88ec58ab)
 I noticed this is due to the different length of my generated vocab-mimic-VQA.json (150 words in total) and your version (147 words in total).
![image](https://github.com/Holipori/EKAID/assets/37804732/b2a174d9-e4c7-4173-ba2b-a09b82dcd930)
![image](https://github.com/Holipori/EKAID/assets/37804732/4f5a85f6-2af4-4eb2-a21a-62144e4dd5fe)
Also, I found my generated  _mimic_pair_questions_temp.csv_ only has 469915 items, should this be 700703 as the paper?
![image](https://github.com/Holipori/EKAID/assets/37804732/a52b9cd1-b3de-45a2-9b38-74cad794aa74)
Dose this mean that my generated dataset encounters some problems, so it is different with your version?

**Comment by Kingofolk** [comment]
Besides, when I run the command python trian_mimic.py --graph all --use_wandb False --eval_target val. The follwoing exception was thrown out when evaluation. It seems coming from the unpair of the val image and the annotations.
![image](https://github.com/Holipori/EKAID/assets/37804732/4c614d1b-b032-4ee5-821c-5c685a342bfc)
Could you please tell me whether this problem comes from my problematic dataset？ 
Thanks a lot. Sincerely appreicate to your patience.
Best
Kingofolk

**Comment by Holipori** [comment]
> Thank you for your reply. When I run mimic_test.py. A new error happened. ![image](https://user-images.githubusercontent.com/37804732/260680868-afeee652-d600-4684-afe1-f91a88ec58ab.png) I noticed this is due to the different length of my generated vocab-mimic-VQA.json (150 words in total) and your version (147 words in total). ![image](https://user-images.githubusercontent.com/37804732/260681252-b2a174d9-e4c7-4173-ba2b-a09b82dcd930.png) ![image](https://user-images.githubusercontent.com/37804732/260681512-4f5a85f6-2af4-4eb2-a21a-62144e4dd5fe.png) Also, I found my generated _mimic_pair_questions_temp.csv_ only has 469915 items, should this be 700703 as the paper? ![image](https://user-images.githubusercontent.com/37804732/260763451-a52b9cd1-b3de-45a2-9b38-74cad794aa74.png) Dose this mean that my generated dataset encounters some problems, so it is different with your version?

Hi, the mismatch between your generated vocabulary size and our provided one is due to the constant improvement of our dataset. We have added three more keywords in "level_lib.csv" and "type_lib.csv". However,  our model was trained on an older version of the dataset, which is the one uploaded to Physionet and is currently under review. To use the provided checkpoint, please delete these three keywords, which are "trace, loculated, hazy", and generate the dataset again. Also, please directly use our version of the vocab-mimic-VQA.json file.
I just updated the "level_lib.csv" and "type_lib.csv" in our repository. Thank you for bringing this up!

Regarding the issue of generated dataset size, could you please send me your generated files “mimic_all.csv”, and “mimic_pair_questions_temp.csv”? This will help me identify the problem. My email address is xxh4034@mavs.uta.edu.
Also, what's the length of your generated "all_diseases.json"? Is it 227835?

**Comment by Holipori** [comment]
> Besides, when I run the command python trian_mimic.py --graph all --use_wandb False --eval_target val. The follwoing exception was thrown out when evaluation. It seems coming from the unpair of the val image and the annotations. ![image](https://user-images.githubusercontent.com/37804732/260682554-4c614d1b-b032-4ee5-821c-5c685a342bfc.png) Could you please tell me whether this problem comes from my problematic dataset？ Thanks a lot. Sincerely appreicate to your patience. Best Kingofolk

Hi, this problem is due to a small bug that causes a mismatch between the ground-truth answer set and the used validation set. You can simply use the "--eval_target test" command when training to avoid this bug. To make the code work perfectly in any scenario, line 117 in train_mimic.py needs to be changed to `val_dataset, val_loader = create_dataset(cfg, 'test' if args.eval_target == 'test' else 'val')` 
I have updated this in the newer version of the code.

**Comment by Holipori** [comment]
Thanks for sending me your generated dataset. The size difference is also because of the different versions of the dataset, resulting from the sample rate change in the code. You can simply change line 1180 into `def question_gen(less_yes_no=False):` to generate a similar size to our provided dataset. However, please keep in mind that this still won't generate exactly the same dataset as we provided due to the randomness. I hope this can solve your problem! Best of luck with your research!

**Comment by Kingofolk** [comment]
Hi, I have run the _mimic_test.py_ file with your _vocab_mimic_VQA.json_, the following exception occurs:
![image](https://github.com/Holipori/EKAID/assets/37804732/5127cd50-3d66-4e8f-a078-c057d290d18c)
It seems that the index is out of range.
Also, when I run the mimic_train.py, the same error occurs, as following:
![image](https://github.com/Holipori/EKAID/assets/37804732/af053671-7390-4744-9b70-c6813a4b3914)
Here the mimic-pair_questions.csv fils contains 700703 items. 
Wish you have a nice day and thanks a lot for your help.
Best regards

**Comment by afasijbfk** [comment]
hi, 
I noticed that in requirements.txt, some packages don't contain a version number, but have @files:... . Please, have you encountered this problem when installing requirements.txt file?
![20230922095657](https://github.com/Holipori/EKAID/assets/102459451/5a43c313-230f-4ba4-a951-d4845916f410)
![image](https://github.com/Holipori/EKAID/assets/102459451/8dd53d52-341b-43af-b52e-2222dc61dd1b)

How did you solve the problem? Thanks a lot, looking forward to your reply.

**Comment by Holipori** [comment]
Hi, 
I just updated a cleaner version of the requirements.txt file. Could you try to use this new one?

**Comment by LLR999** [comment]
> Hi! Sorry for bothering you again. I encoountered some problems when running the 'python mimic_test.py _path to checkpoint_' command. The error is as following: ![image](https://private-user-images.githubusercontent.com/37804732/259988666-0ec55cf4-646e-437b-b88b-558bfd87e73a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE2NzIxNzksIm5iZiI6MTcwMTY3MTg3OSwicGF0aCI6Ii8zNzgwNDczMi8yNTk5ODg2NjYtMGVjNTVjZjQtNjQ2ZS00MzdiLWI4OGItNTU4YmZkODdlNzNhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA0VDA2Mzc1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ4M2IyNDk0ODJlOWQ5YzkzMzNiNTI5YmY0Njg3YjQxMDU5ZjBhNjBjY2RkNTVmNWNiMDFkZDMzNjM2ZTllYzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QiJOJwI6PVBr4WQa3owVbIGDRhhJ6zhbETysR5RdqrY) Should I add some lines to unify their type in the following section or could you please offer me the optimal solution? ![image](https://private-user-images.githubusercontent.com/37804732/259988884-1e12de53-99c8-4b10-8124-3afe81cdf9ea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE2NzIxNzksIm5iZiI6MTcwMTY3MTg3OSwicGF0aCI6Ii8zNzgwNDczMi8yNTk5ODg4ODQtMWUxMmRlNTMtOTljOC00YjEwLTgxMjQtM2FmZTgxY2RmOWVhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA0VDA2Mzc1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkZDA3ZjQ3YWI5ZDg0NjhiZDAwODdiNjIwMTAxN2U4ZDhiZGY2NGU2YWU3MTkwMDc0YTZkNDQ4ODExNTQ4NTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.a5_hvUu1c1iLvAEHBbQLeWINTfy-mMw7mNQRDUb7wv8)
> 
> Besides, I do not know whether this problem is due to the inproporiate package version. I also have problems in installing the required libs from the requirement.txt. Like this: ![image](https://private-user-images.githubusercontent.com/37804732/259989219-0313ee83-3815-4786-8dfa-a26a8f0301ae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE2NzIxNzksIm5iZiI6MTcwMTY3MTg3OSwicGF0aCI6Ii8zNzgwNDczMi8yNTk5ODkyMTktMDMxM2VlODMtMzgxNS00Nzg2LThkZmEtYTI2YThmMDMwMWFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA0VDA2Mzc1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyNDdhMTljZmIyNGQ0ZDIwYzliOGQ5MmFmOTU1OTM3MTlkMTRjNTQ5ZGI1NDg5YzM2N2FmZjQ0NWFkNmI0MzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cYyTIhU_l03tXSPscZ93L3f8X8vQIyqB4X6cB9Hd_cA) So I installed all the needed libs manually. Looking forward to your reply. Thank you very much and wich you have a nice day. Kingofolk

I am also an postgraduate student in china and interested in this newly dataset,can  you share it with me ?

**Comment by Holipori** [comment]
> I am also an postgraduate student in china and interested in this newly dataset,can you share it with me ?

Hi, the dataset can be accessed from here: https://physionet.org/content/medical-diff-vqa/1.0.0/
